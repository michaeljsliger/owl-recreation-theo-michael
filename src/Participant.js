import React, {Component} from 'react';





class Participant extends Component {
    render(){
        return(
            <div className="sessionParticipant">
            
                    <img className="user-avatar" src={this.props.avatar} alt="avatar"/>
            

                <div className= "participant-info" id={this.props.id}>
                    <h6 className= "particpant-name">{this.props.name}</h6>
                </div>
                <div className="participant-status"> 

                </div>
            </div>  

        
         
        )

    }
}
export default Participant;