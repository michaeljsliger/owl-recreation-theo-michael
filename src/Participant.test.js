import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import Participant from './Participant';
import renderer from 'react-test-renderer';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

describe('Participant Box Branch Testing', () => {
  // smoke test

  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Participant />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  // snapshot test
  it('should render correctly', () => {
    const tree = renderer
          .create(<Participant />)
          .toJSON();
    expect(tree).toMatchSnapshot();
  })
})