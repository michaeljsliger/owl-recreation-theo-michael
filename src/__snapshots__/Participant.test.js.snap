// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant Box Branch Testing should render correctly 1`] = `
<div
  className="sessionParticipant"
>
  <img
    alt="avatar"
    className="user-avatar"
  />
  <div
    className="participant-info"
  >
    <h6
      className="particpant-name"
    />
  </div>
  <div
    className="participant-status"
  />
</div>
`;
